packages/build-tools/src/context.ts

```diff
@@ -123,6 +126,13 @@ export class BuildContext<TJob extends Job = Job> {
       ? [this.buildExecutablesDirectory, this._env.PATH].join(':')
       : this.buildExecutablesDirectory;
 
+    this._isLocal = this._env.EAS_BUILD_RUNNER !== 'eas-build';
+
+    this._isUsingCache =
```

Comment:
Let's not add this and instead add an early return for if `!isLocal` in cache-related phases? this logic has two copies, one for saving and one for restoring